### DISCUSSION TITLE
What is the source of this data?

### DISCUSSION THREAD

**Discussion by DartPads** [codereverser/casparser-isin#4] [Q&A]
Thank you for your hard work. If you don't mind can you please tell what is the source of this data. Thanks again

**Comment by codereverser** [comment]
No problem! The db is built by merging data from multiple sources . 
- [BSE StarMF Master data](https://bsestarmf.in/RptSchemeMaster.aspx)  - scheme details with RTA code 
- [AMFI NAV](https://www.amfiindia.com/spages/NAVAll.txt) - AMFI code 
- [Quandl AMFI API Metadata](https://www.quandl.com/data/AMFI-Association-of-Mutual-Funds-in-India) - AMFI code

**Reply by DartPads** [reply]
WOW, good to know I'm not the only one who is doing a similar exercise. I've a couple follow up questions (1) How often do you update the database (2) Where did you get enough CAS pdf's to test your code? I've only one PDF to test 😞.

Thanks again and I really appreciate your time.

**Reply by codereverser** [reply]
1) The main package (including db) in pypi is updated rarely . but I try to update the database twice a week at least. The process is manual right now, but I'm working on a GitHub scheduled action to build and deploy it automatically daily or every other day or so. 

2) I also have access to only a couple of PDF files, mainly sourced from friends and family. The majority of testing and bug fixing is crowd-sourced via github issues in the main [casparser](https://github.com/codereverser/casparser/issues) repo
